### PR TITLE
[Feature] DataFrame export for MSSpectrum and MSChromatogram

### PR DESCRIPTION
**Pull request by axelwalter**
## Description

For the upcoming pyopenms visualization module we need more OpenMS objects exported as DataFrames. Here, MSSpectrum and MSChromatogram get new methods for DataFrame export.

TODO: add tests

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

## **Type**
enhancement, tests

___

## **Description**
- Added DataFrame export capabilities to `MSSpectrum` and `MSChromatogram` classes in pyOpenMS.
- Implemented new classes `_MSSpectrumDF` and `_MSChromatogramDF` to handle the DataFrame conversions.
- Enhanced the module and class naming within the documentation to align with Sphinx autodoc requirements.
- Created comprehensive tests to validate the new DataFrame functionalities and ensure data integrity.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>_dataframes.py</strong><dd><code>Implement DataFrame Export for MSSpectrum and MSChromatogram</code></dd></summary>
<hr>
      
src/pyOpenMS/pyopenms/_dataframes.py

<li>Added new classes <code>_MSSpectrumDF</code> and <code>_MSChromatogramDF</code> for DataFrame <br>export functionalities.<br> <li> Added DataFrame export methods to <code>MSSpectrum</code> and <code>MSChromatogram</code>.<br> <li> Updated class module and name attributes to ensure correct <br>documentation generation.<br> <li> Added utility function <code>_add_meta_values</code> to append meta values to <br>DataFrames.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7481/files#diff-430a5767c0a5065fbde4c38a60286b16598929f6f396a775ac7218b7da491379">+165/-1</a>&nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Tests
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test000.py</strong><dd><code>Add Tests for MSSpectrum and MSChromatogram DataFrame Exports</code></dd></summary>
<hr>
      
src/pyOpenMS/tests/unittests/test000.py

<li>Added tests for the new DataFrame export functionalities of <code>MSSpectrum</code> <br>and <code>MSChromatogram</code>.<br> <li> Ensured that the DataFrame contains correct values and structure as <br>expected.<br>

</details>
    

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7481/files#diff-9b64f80ffadfaff913d4cd4a67264c1b950fc47ad83ebf98bcaf1431d6479354">+42/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: src/pyOpenMS/pyopenms/_dataframes.py [hunk 1]
```diff
@@ -608,5 +611,145 @@ def update_scores_from_df(peps: List[_PeptideIdentification], df : _pd.DataFrame
...
+        Returns a DataFrame representation of the MSSpectrum.
+
+        Args:
+            export_meta_values (bool): Whether to export meta values.
+            export_peptide_identifications (bool): Whether to export peptide identifications.
+
+        Returns:
+            pd.DataFrame: DataFrame representation of the MSSpectrum.
+        """
+        def extract_peak_data(s: _MSSpectrum):
```
Comment (timosachsenberg) [inline src/pyOpenMS/pyopenms/_dataframes.py]: note that we have already some numpy code: https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/addons/MSSpectrum.pyx#L6

**Comment by timosachsenberg** [comment]
can you add more information for others what the exact goals are? This will make it easier to check for performance / usability considerations.

File: src/pyOpenMS/pyopenms/_dataframes.py (hunk 1 above)
Comment (axelwalter) [reply src/pyOpenMS/pyopenms/_dataframes.py]: good point!

File: src/pyOpenMS/pyopenms/_dataframes.py [hunk 2]
```diff
@@ -134,6 +135,8 @@ def get_df(self):
         return _pd.concat([self.get_metadata_df(), self.get_intensity_df()], axis=1)
 
 ConsensusMap = _ConsensusMapDF
+ConsensusMap.__module__ = _ConsensusMap.__module__
```
Comment (jpfeuffer) [inline src/pyOpenMS/pyopenms/_dataframes.py:142]: Awesome thanks. Can you add a comment here or in the docs on why this is necessary and in which case it is needed?

**Comment by timosachsenberg** [comment]
Can you also double check with the code we use for ion mobility? I think @jpfeuffer added some code that is able to export it.

**Comment by jpfeuffer** [comment]
I think it was a contribution where i helped with. I recently made it a bit faster but I think that improvement is not merged yet.

File: src/pyOpenMS/pyopenms/_dataframes.py [hunk 3]
```diff
@@ -608,5 +630,260 @@ def update_scores_from_df(peps: List[_PeptideIdentification], df : _pd.DataFrame
...
+        if export_meta_values:
+            df = _add_meta_values(df, self)
+
+        return df
+
+MSSpectrum = _MSSpectrumDF
+MSSpectrum.__module__ = _MSSpectrum.__module__
+MSSpectrum.__name__ = 'MSSpectrum'
+
+class _ChromatogramType(_Enum):
```
Comment (timosachsenberg) [inline src/pyOpenMS/pyopenms/_dataframes.py:725]: I wonder if these are not already defined somewhere else?

Comment (timosachsenberg) [reply src/pyOpenMS/pyopenms/_dataframes.py:725]: they are not specific to dataframes

**Review by timosachsenberg — APPROVED** [review]

**Pull request merged**